src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -200,10 +208,59 @@ internal int SerializeLogRecord(LogRecord logRecord)
             var name = logRecord.CategoryName;
 
             // If user configured explicit TableName, use it.
-            if (this.m_tableMappings == null || !this.m_tableMappings.TryGetValue(name, out var eventName))
+            if (this.m_tableMappings == null || (!this.m_tableMappings.TryGetValue(name, out var eventName) && !this.shouldPassThruTableMappings))
             {
                 eventName = this.m_defaultEventName;
             }
+            else if (this.shouldPassThruTableMappings && eventName == null)
+            {
+                char[] tempArr = new char[name.Length];
+                int readIdx = 0;
+                int writeIdx = 0;
+                while (readIdx < name.Length)
+                {
+                    if (readIdx == 0)
```

Comment:
You could move this check outside of the loop to avoid checking it again and again.